target/linux/ramips/mt7621/base-files/lib/upgrade/platform.sh

```diff
@@ -53,6 +53,10 @@ platform_do_upgrade() {
 	xiaomi,mir3p)
 		nand_do_upgrade "$1"
 		;;
+	iodata,wn-dx1167r)
+		iodata_mstc_upgrade_prepare
```

Comment:
How is this working when you do not include the other file?

Reply:
If so, regardless of the os-image bank used to boot at that time, sysupgrade image is written to the 1st bank ("ubi") and restarted without changing ```bootnum``` to "1" and  ```debugflag``` enabled.

If stock firmware booted from 1st bank ("ubi"):
1. write OpenWrt initramfs image to 2nd bank ("firmware_2") and change ```bootnum``` to "2" on stock firmware
  - bootnum: 1 -> 2
  - debugflag: 0
2. boot OpenWrt initramfs image from 2nd bank, perform sysupgrade (sysupgrade image will be written to 1st bank)
  - bootnum: 2
  - debugflag: 0
3. boot OpenWrt initramfs image from 2nd bank

If stock firmware booted from 2nd bank ("firmware_2"):
1. write OepnWrt initramfs image to 1st bank ("ubi") and change ```bootnum``` to "1" on stock firmware
  - bootnum: 2 -> 1
  - debugflag: 0
2. boot OpenWrt initramfs image from 1st bank, perform sysupgrade (sysupgrade image will be written to 1st bank)
  - bootnum: 1
  - debugflag: 0
3. boot OpenWrt sysupgrade image from 1st bank

Reply:
My question was actually meant much easier: Should we include ". /lib/upgrade/iodata.sh" at the top?

Reply:
Ah, I see. All scripts in /lib/upgrade/ are loaded by ["do_stage2" script](https://github.com/openwrt/openwrt/blob/5f5ec7660ca26d5637c77f45d88c2e2eb35d3bb1/package/base-files/files/lib/upgrade/do_stage2#L5).

Reply:
Ah, okay. I'd personally still add an additional include in /lib/upgrade/platform.sh, but I see that it's not required.
So, just tell me what you prefer and I will merge it that way.

Reply:
Thank you, I want to keep this state to align with others.